packages/pyright-internal/src/tests/lsp/languageServer.ts

```diff
@@ -325,7 +326,7 @@ class ServerStateManager {
             if (serverIndex >= 0) {
                 try {
                     instance.disposables[serverIndex].dispose();
-                    instance.disposables = instance.disposables.splice(serverIndex, 1);
+                    instance.disposables.splice(serverIndex, 1);
```

Comment:
This was a bug in the cleanup. `splice` doesn't return the modified array, it modifies it inline and returns the deleted items. This bug was preventing the connection from being disposed, meaning all test servers were still listening to messages.